### PR TITLE
Add option to disable JavaScript routers

### LINKED ISSUE #12262: [Feature] Add option to disable Javascript routers
I did not found in [documentation](https://www.playframework.com/documentation/3.0.x/ScalaJavascriptRouting) how to turn off generation of JavaScript routers
It would be nice to have that, for decreasing of unused generated code.
Example: REST client (`PlayService`) without web interface, so JavaScript routers will never be used here.

### PR DESCRIPTION
**Pull request by PromanSEW**
# Pull Request Checklist

* [X] Have you read [How to write the perfect pull request](https://github.blog/2015-01-21-how-to-write-the-perfect-pull-request/)?
* [X] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [X] Have you referenced any issues you're fixing using [commit message keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)?
* [X] Have you added copyright headers to new files?
* [X] Have you checked that both Scala and Java APIs are updated?
* [ ] Have you updated the documentation for both Scala and Java sections?
* [x] Have you added tests for any changed functionality?

# Helpful things

## Fixes

Fixes #12262

## Purpose

Add option to disable JavaScript routers

### CODE REVIEW

**Comment by mkurz** [comment]
Thanks, I will do some local testing when I find time. Actually you said you don't need that urgently because you can just disable all reverse routes generate now and that solves your problem, yes?

File: dev-mode/play-routes-compiler/src/main/scala/play/routes/compiler/RoutesGenerator.scala [hunk 1]
```diff
@@ -68,15 +68,18 @@ object InjectedRoutesGenerator extends RoutesGenerator {
     }
 
     val reverseRoutesFiles = if (task.reverseRouter) {
-      generateReverseRouters(sourceInfo, namespace, task.additionalImports, routes, task.namespaceReverseRouter) ++
-        generateJavaScriptReverseRouters(
+      var reverseRouters =
```
Comment (PromanSEW) [inline dev-mode/play-routes-compiler/src/main/scala/play/routes/compiler/RoutesGenerator.scala:71]: Should I rewrite this as below?
```scala
generateReverseRouters(...) ++
  if (task.jsReverseRouter) {
  ...
  } else {
    Nil
  } ++ generateJavaWrappers(sourceInfo, namespace, rules, task.namespaceReverseRouter)
```

Comment (mkurz) [reply dev-mode/play-routes-compiler/src/main/scala/play/routes/compiler/RoutesGenerator.scala:71]: Yes, that's maybe nicer. Thanks!

**Comment by PromanSEW** [comment]
@mkurz yes, I don't need this change for now, disabling all routes was in fact I am looking for

File: dev-mode/play-routes-compiler/src/main/scala/play/routes/compiler/RoutesGenerator.scala [hunk 2]
```diff
@@ -69,14 +69,22 @@ object InjectedRoutesGenerator extends RoutesGenerator {
 
     val reverseRoutesFiles = if (task.reverseRouter) {
       generateReverseRouters(sourceInfo, namespace, task.additionalImports, routes, task.namespaceReverseRouter) ++
-        generateJavaScriptReverseRouters(
+        (if (task.jsReverseRouter) {
```
Comment (PromanSEW) [inline dev-mode/play-routes-compiler/src/main/scala/play/routes/compiler/RoutesGenerator.scala:72]: IDEA shows error without braces around if

**Comment by PromanSEW** [comment]
@mkurz can you review please?

**Review by mkurz — APPROVED** [review]
Thanks! I pushed two commit:
- First I made sure we stay compatible with previous releases. This is not 100% necessary here because MiMa does not complain, that is because this is an sbt plugin and we do not guarantee compatibility for them, also it's meant for interal use basically, however I think it does not hurt and in the very unlikely case someone is using `RoutesCompilerTask` it won't break their code. Actually I would have added that constructor no matter if we include this pr in a patch release or a "minor"/"major" release anyway.
- The second commit is much more important, because you kept generating a `class javascript` within the java wrappers, which, when running a real world app with javascript reverse routers disabled, fails compiling, because of course there is no javascript package anymore....

**Comment by mkurz** [comment]
@Mergifyio backport 2.9.x

**Pull request merged**